src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -323,6 +326,42 @@ public void testCredentialReplacing() throws Exception {
         assertTrue(identity3.verifyEvidence(new PasswordGuessEvidence("secretPassword".toCharArray())));
     }
 
+    @Test
+    public void testIterating() throws Exception {
+        FileSystemSecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(), 1);
+        ModifiableRealmIdentity identity1 = securityRealm.getRealmIdentityForUpdate("firstUser", null, null);
+        identity1.create();
+        ModifiableRealmIdentity identity2 = securityRealm.getRealmIdentityForUpdate("secondUser", null, null);
+        identity2.create();
+        Iterator<ModifiableRealmIdentity> iterator = securityRealm.getRealmIdentityIterator();
+
+        int count = 0;
+        while(iterator.hasNext()){
+            ModifiableRealmIdentity identity = iterator.next();
+            Assert.assertTrue(identity.exists());
+            count++;
+        }
+
+        Assert.assertEquals(2, count);
+        getRootPath(); // will fail on windows if iterator not closed correctly
+    }
+
+    @Test
+    public void testPartialIterating() throws Exception {
+        FileSystemSecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(), 1);
+        ModifiableRealmIdentity identity1 = securityRealm.getRealmIdentityForUpdate("firstUser", null, null);
+        identity1.create();
+        ModifiableRealmIdentity identity2 = securityRealm.getRealmIdentityForUpdate("secondUser", null, null);
+        identity2.create();
+        Iterator<ModifiableRealmIdentity> iterator = securityRealm.getRealmIdentityIterator();
```

Comment:
I am thinking of at least one test should use a try-with-resources style - although the current test shows it can be used as a normal iterator it needs to be cast to close it.

Also would be nice to see a test where close is called after the end is reached, although you may be handling that automatically we should be sure it will silently work as once something implements Closeable anything using it should close it regardless at the end.